tfhe/src/safe_deserialization.rs

```diff
@@ -159,76 +159,47 @@ mod test_integer {
     use crate::prelude::*;
     use crate::safe_deserialization::{safe_deserialize_conformant, safe_serialize};
     use crate::shortint::parameters::{
-        PARAM_MESSAGE_1_CARRY_1_KS_PBS, PARAM_MESSAGE_2_CARRY_2_KS_PBS,
-        PARAM_MESSAGE_3_CARRY_3_KS_PBS,
+        PARAM_MESSAGE_2_CARRY_2_KS_PBS, PARAM_MESSAGE_3_CARRY_3_KS_PBS,
     };
     use crate::{
-        CompactFheUint8, CompactFheUint8List, CompactFheUint8ListConformanceParams,
-        CompactPublicKey, FheUint8ConformanceParams,
+        set_server_key, CompactCiphertextList, CompactCiphertextListConformanceParams,
+        CompactPublicKey, FheUint8,
     };
-    #[test]
-    fn safe_deserialization_ct() {
-        let (client_key, server_key) = generate_keys(ConfigBuilder::default().build());
-
-        let public_key = CompactPublicKey::new(&client_key);
-
-        let msg = 27u8;
-
-        let ct = CompactFheUint8::try_encrypt(msg, &public_key).unwrap();
-
-        let mut buffer = vec![];
-
-        safe_serialize(&ct, &mut buffer, 1 << 40).unwrap();
-
-        let params = FheUint8ConformanceParams::from(PARAM_MESSAGE_1_CARRY_1_KS_PBS);
-        assert!(safe_deserialize_conformant::<CompactFheUint8>(
-            buffer.as_slice(),
-            1 << 20,
-            &params
-        )
-        .is_err());
-
-        let params = FheUint8ConformanceParams::from(&server_key);
-        let ct2 =
-            safe_deserialize_conformant::<CompactFheUint8>(buffer.as_slice(), 1 << 20, &params)
-                .unwrap();
-
-        let dec: u8 = ct2.expand().decrypt(&client_key);
-        assert_eq!(msg, dec);
-    }
-
     #[test]
     fn safe_deserialization_ct_list() {
-        let (client_key, server_key) = generate_keys(ConfigBuilder::default().build());
+        let (client_key, sks) = generate_keys(ConfigBuilder::default().build());
+        set_server_key(sks);
 
         let public_key = CompactPublicKey::new(&client_key);
 
         let msg = [27u8, 10, 3];
 
-        let ct_list = CompactFheUint8List::try_encrypt(&msg, &public_key).unwrap();
+        let ct_list = CompactCiphertextList::builder(&public_key)
+            .push(27u8)
+            .push(10u8)
+            .push(3u8)
```

Comment:
question, does it try to fit the type in the smallest FheUint type ?

Reply:
FheType::Uint32(u32) e.g. ? send the payload via the enum and the vairant gives the type ?

Reply:
Hum it would not solve the problem for things like FheUint{2,4,6,10,12,..} because we don't have u2, u4, etc

Also we already have an enum `FheType` that does not have payload, it is used for example to query the type that is stored in a particular slot (index) of the compact list

And neither options of adding a payload to that enum (which then this payload won't always be used), or adding a different enum that has the payload seems particularly appealing

Also the `FheType` enum may be expanded if we add vectors/tensor types (unless we decide to change the way we describe our types)

Reply:
indeed, question is do we keep those "bastard" FheTypes though ?

Reply:
ok, could be good to know if we can ditch those types

Understood for the dynamic types as fallback

static types sound like they could work but would likely require some macro to make it work, in any case for BC it's likely not useful as the C API would need an adaptation